### PR TITLE
Checkphish: New App - Initial Contribution

### PR DESCRIPTION
**Pull request by dfederschmidt**
#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
- [x] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Added initial release note in subfolder

## Other information
- API does not offer suitable endpoint for connectivity testing (only offers to detonate a URL, which would reduce quota)

### CODE REVIEW

**Comment by dhwanis-crest** [comment]
Hi @dfederschmidt , Could you provide access rights to me for this repo?

**Comment by dhwanis-crest** [comment]
Hi @dfederschmidt ,
As per the existing code, the detonate_url action passes with invalid parameters as the api returns 200 status code. But, it also provides an error message. Should the action pass or fail? If it should fail then on what basis we should fail the action?
image

**Comment by dfederschmidt** [comment]
> Hi @dfederschmidt , As per the existing code, the detonate_url action passes with invalid parameters as the api returns 200 status code. But, it also provides an error message. Should the action pass or fail? If it should fail then on what basis we should fail the action? image

I think it would be sensible to fail the action in case there is a field non-empty field `errorMessage` returned as part of the response.

File: checkphish_connector.py [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+                action_result.set_status(
+                    phantom.APP_ERROR,
+                    "Error Connecting to server. Details: {0}".format(self._get_error_message_from_exception(e)),
+                ),
+                resp_json,
+            )
+
+        return self._process_response(r, action_result)
+
+    def _handle_test_connectivity(self, param):
```
Comment (joshquerinphantom) [inline checkphish_connector.py:200]: I saw that the readme says this app does not implement test_connectivity. Why not?

Comment (dhwanis-crest) [reply checkphish_connector.py:200]: As this is a third-party community app, this is how the contributor has kept it. We don’t make changes in the community apps as we don't have a positive asset for that.

Comment (joshquerinphantom) [reply checkphish_connector.py:200]: Ah understood.  I do not know the history, but I think in general we should push back on community submission to include test_connectivity as it is vital for most customers (asset health page, for example). 

If this has been raised and approved, though, then I will approve the PR

Comment (dhwanis-crest) [reply checkphish_connector.py:200]: Hi @dfederschmidt , can you please confirm whether you are planning to implement the test-connectivity action or not?

Comment (dfederschmidt) [reply checkphish_connector.py:200]: @joshquerinphantom the original omission of the `test connectivity` action is outlined in the original MR description.

> API does not offer suitable endpoint for connectivity testing (only offers to detonate a URL, which would reduce quota)

The API does not offer an authenticated endpoint that we can call other than the one called for `detonate url`, reducing quota. I suppose we can call the endpoint for a well-known URL (eg. `google.com` or `checkphish.ai`) and document that running the test connectivity action is not "free".
I'll push that addition in a bit.

Comment (dfederschmidt) [reply checkphish_connector.py:200]: @dhwanis-crest Please review https://github.com/splunk-soar-connectors/checkphish/pull/1/commits/9ffa524314c7dd972b9330eb2fcad04a8dd422cd

Comment (dhwanis-crest) [reply checkphish_connector.py:200]: @dfederschmidt Reviewed and verified.

**Review by joshquerinphantom — DISMISSED** [review]

**Comment by joshquerinphantom** [comment]
Resolved thread and approved changes

File: checkphish_connector.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+        elif action_id == "detonate_url":
+            ret_val = self._handle_detonate_url(param)
+
+        return ret_val
+
+    def initialize(self):
+        self._state = self.load_state()
+
+        self._state = self.load_state()
```
Comment (paulnawat) [inline checkphish_connector.py]: Is this typo? we only need to load once, right?

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+beautifulsoup4==4.9.1
+requests==2.25.0
```
Comment (paulnawat) [inline requirements.txt:2]: Should we add the other 5 dependencies from `wheel` here? see [your wheel](https://github.com/splunk-soar-connectors/checkphish/pull/1/files#diff-8326d0c384e1236edb93244e2b1f11192224aa60db0b125515430a10f0733953R275)

File: checkphish_connector.py [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+        )
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _process_response(self, r, action_result):
+        # store the r_text in debug data, it will get dumped in the logs if the action fails
+        if hasattr(action_result, "add_debug_data"):
+            action_result.add_debug_data({"r_status_code": r.status_code})
+            action_result.add_debug_data({"r_text": r.text})
+            action_result.add_debug_data({"r_headers": r.headers})
```
Comment (paulnawat) [inline checkphish_connector.py:139]: Would `self.debug_print('status code: {}'.format(r.status_code))` make it easier to log? or access the log after?

**Review by paulnawat — CHANGES_REQUESTED** [review]
Changes look great 👍 
I left a few questions / comments.

File: requirements.txt (hunk 3 above)
Comment (dhwanis-crest) [reply requirements.txt:2]: No, those wheels files are automatically installed with these two as they are dependent.

Comment (paulnawat) [reply requirements.txt:2]: ah I see. Thanks

File: checkphish_connector.py (hunk 4 above)
Comment (dhwanis-crest) [reply checkphish_connector.py:139]: Adding the response data in action will only dump in case of action failure. Hence, not using self.debug_print.

Comment (paulnawat) [reply checkphish_connector.py:139]: That's fair. Thanks!

File: checkphish_connector.py (hunk 2 above)
Comment (paulnawat) [reply checkphish_connector.py]: I saw the remove commit. Resolving this.

Comment (dhwanis-crest) [reply checkphish_connector.py]: Yes, fixed

**Review by paulnawat — APPROVED** [review]

**Pull request merged**